application/models/Token.php

```diff
@@ -366,7 +383,11 @@ public function rules()
             array('emailstatus', 'default', 'value' => 'OK'),
         );
         foreach (decodeTokenAttributes($this->survey->attributedescriptions) as $key => $info) {
-                $aRules[] = array($key, 'LSYii_Validators', 'except'=>'FinalSubmit');
+            $aRules[] = array(
+                $key, 'filter',
+                'filter' => array(self::class, 'sanitizeAttribute'),
+                'except'=>'FinalSubmit'
```

Comment:
I muts add `'except'=>'FinalSubmit'` for firstname or lastname too ? It's a sanitizer, not a checker.